time-macros/src/lib.rs

```diff
@@ -89,3 +91,114 @@ pub fn format_description(input: TokenStream) -> TokenStream {
         DESCRIPTION
     }}
 }
+
+fn make_serde_serializer_module(
+    mod_name: proc_macro::Ident,
+    items: impl to_tokens::ToTokens,
+    formattable: TokenStream,
+) -> TokenStream {
+    let serialize_fns = quote! {
+        pub fn serialize<S: ::serde::Serializer>(
+                datetime: &#(formattable.clone()),
+                serializer: S,
+            ) -> Result<S::Ok, S::Error> {
+                use ::serde::Serialize;
+                datetime.format(&DESCRIPTION)
+                    .map_err(::time::error::Format::into_invalid_serde_value::<S>)?
+                    .serialize(serializer)
+        }
+
+        pub fn deserialize<'a, D: ::serde::Deserializer<'a> >(
+                deserializer: D
+        ) -> Result<#(formattable.clone()), D::Error> {
+            use ::serde::Deserialize;
+                #(formattable.clone())::parse(<&str>::deserialize(deserializer)?, &DESCRIPTION)
+                    .map_err(time::error::Parse::to_invalid_serde_value::<D>)
+        }
+    };
+    let option_serialize_fns = quote! {
+        pub fn serialize<S: ::serde::Serializer>(
+            option: &Option<#(formattable.clone())>,
+            serializer: S,
+        ) -> Result<S::Ok, S::Error> {
+            use ::serde::Serialize;
+            option.map(|datetime| datetime.format(&DESCRIPTION))
+                    .transpose()
+                    .map_err(::time::error::Format::into_invalid_serde_value::<S>)?
+                    .serialize(serializer)
+        }
+
+        pub fn deserialize<'a, D: ::serde::Deserializer<'a> >(
+            deserializer: D
+        ) -> Result<Option<#(formattable.clone())>, D::Error> {
+            use ::serde::Deserialize;
+            Option::<&str>::deserialize(deserializer)?
+                .map(|string| #(formattable)::parse(string, &DESCRIPTION))
+                .transpose()
+                .map_err(time::error::Parse::to_invalid_serde_value::<D>)
+        }
+    };
+
+    quote! {
+        mod #(mod_name) {
+            const DESCRIPTION: &[::time::format_description::FormatItem<'_>] = &[#(items)];
+
+            #(serialize_fns)
+
+            pub mod option {
+                use super::DESCRIPTION;
+
+                #(option_serialize_fns)
+            }
+        }
+    }
+}
+
+fn declare_format_string(input: TokenStream, formattable: TokenStream) -> TokenStream {
+    let mut tokens = input.into_iter();
+    // First, an identifier (the desired module name)
+    let mod_name = match tokens.next() {
+        Some(TokenTree::Ident(ident)) => ident,
+        Some(tree) => {
+            return Error::UnexpectedToken { tree }.to_compile_error_standalone();
+        }
+        None => return Error::UnexpectedEndOfInput.to_compile_error_standalone(),
+    };
+    // Followed by a comma
+    match tokens.next() {
+        Some(tree) => {
+            if let TokenTree::Punct(ref punct) = tree {
+                if punct.as_char() != ',' {
```

Comment:
```suggestion
            if matches!(tree, TokenTree::Punct(ref punct) if punct.as_char() != ',') {
```

Also might be able to omit the `.as_char()`. I forget if it was stabilized before or after the MSRV.

Reply:
Looks like the [`PartialEq<char> for Punct`](https://doc.rust-lang.org/stable/proc_macro/struct.Punct.html#impl-PartialEq%3Cchar%3E) was added in 1.50 so we're good